File: source/Assets/Scripts/BulletCollision.js




function OnTriggerEnter2D(obj : Collider2D) {

		var cs = GameObject.Find("Player1");
		//var playerHealth = cs.GetComponent("PlayerHealth");
		//var playerSpawn = cs.GetComponent("PlayerSpawn");

	
		
		var tag = obj.gameObject.tag;
		

		if (tag == "Wall"){
			Network.Destroy(GetComponent(NetworkView).id);
			//playerHealth.TakeDamage(5);
		}
		

		if (tag == "Player" || tag == "Enemy"){
			Network.Destroy(GetComponent(NetworkView).id);
			
			//Destroy (this.gameObject);
			//obj.playerHealth.TakeDamage();
		}
		
		/* if (tag == "Player" | tag == "Enemy"){
			Destroy (this.gameObject);
		} */

	}

File: source/Assets/Scripts/BulletCollision.cs
using UnityEngine;
using System.Collections;

public class BulletCollision : MonoBehaviour {

	private GameObject zObject;
	PlayerHealth playerHealth; 


	void Awake() {
	}


	void OnTriggerEnter2D(Collider2D other) {
		playerHealth = GameObject.Find(other.name).GetComponent<PlayerHealth>();
		zObject = other.gameObject;

		if(other.tag == "Wall") {
			//playerHealth.TakeDamage(5);
			Debug.Log("test");
			Network.Destroy(GetComponent<NetworkView>().viewID);
		}

		if(other.tag == "Player") {
			playerHealth.TakeDamage(5);
			Network.Destroy(GetComponent<NetworkView>().viewID);
		}
	}
}
